ipa-core/src/report/hybrid.rs

```diff
@@ -308,6 +441,282 @@ where
+                impression_report.encap_key_mk()
+            }
+            EncryptedHybridGeneralReport::Conversion(conversion_report) => {
+                conversion_report.encap_key_mk()
+            }
+        }
+    }
+
+    pub fn mk_ciphertext(&self) -> &[u8] {
+        match self {
+            EncryptedHybridGeneralReport::Impression(impression_report) => {
+                impression_report.mk_ciphertext()
+            }
+            EncryptedHybridGeneralReport::Conversion(conversion_report) => {
+                conversion_report.mk_ciphertext()
+            }
+        }
+    }
+
+    pub fn encap_key_btt(&self) -> &[u8] {
+        match self {
+            EncryptedHybridGeneralReport::Impression(impression_report) => {
+                impression_report.encap_key_btt()
+            }
+            EncryptedHybridGeneralReport::Conversion(conversion_report) => {
+                conversion_report.encap_key_btt()
+            }
+        }
+    }
+
+    pub fn btt_ciphertext(&self) -> &[u8] {
+        match self {
+            EncryptedHybridGeneralReport::Impression(impression_report) => {
+                impression_report.btt_ciphertext()
+            }
+            EncryptedHybridGeneralReport::Conversion(conversion_report) => {
+                conversion_report.btt_ciphertext()
+            }
+        }
+    }
+
+    pub fn key_id(&self) -> u8 {
+        match self {
+            EncryptedHybridGeneralReport::Impression(impression_report) => {
+                impression_report.key_id()
+            }
+            EncryptedHybridGeneralReport::Conversion(conversion_report) => {
+                conversion_report.key_id()
+            }
+        }
+    }
+
+    /// ## Errors
+    /// If the report contents are invalid.
+    pub fn from_bytes(mut bytes: Bytes) -> Result<Self, InvalidHybridReportError> {
+        //let first_byte = bytes.next().ok_or(InvalidHybridReportError::Length(0,1))?;
+        //let the_rest = bytes[1..];
+        match bytes[0] {
+            1 => {
+                //let impression_report = EncryptedHybridImpressionReport::<BK, B>::from_bytes(&bytes[1..])?;
+                bytes.advance(1);
+                let impression_report = EncryptedHybridImpressionReport::<BK>::from_bytes(bytes)?;
+                Ok(EncryptedHybridGeneralReport::Impression(impression_report))
+            }
+            0 => {
+                //let conversion_report = EncryptedHybridConversionReport::<V, B>::from_bytes(&bytes[1..])?;
+                bytes.advance(1);
+                let conversion_report = EncryptedHybridConversionReport::<V>::from_bytes(bytes)?;
+                Ok(EncryptedHybridGeneralReport::Conversion(conversion_report))
+            }
+            _ => Err(InvalidHybridReportError::UnknownEventType(bytes[0])),
+        }
+    }
+
+    /// ## Errors
+    /// If decryption of the provided oprf report fails.
+    pub fn decrypt_from_oprf_report_bytes<P, TS>(
```

Comment:
we don't need to keep this long term. it would be great if we could get everything to work without this, but if we want to split that into another PR, that's fine.

Reply:
yeah, I think I'll just remove this once all the oprf is completely disentangled